### PR TITLE
feat(spark): OpenLineage 1.24.2 upgrade

### PR DESCRIPTION
**Pull request by treff7es**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-integration/java/acryl-spark-lineage/build.gradle [hunk 1]
```diff
@@ -123,39 +127,36 @@ shadowJar {
...
   relocate 'com.sun.codemodel', 'io.acryl.shaded.com.sun.codemodel'
   relocate 'com.sun.mail', 'io.acryl.shaded.com.sun.mail'
-  relocate 'com.fasterxml.jackson', 'datahub.spark2.shaded.jackson'
-  relocate 'org.slf4j', 'datahub.spark2.shaded.org.slf4j'
   //
   relocate 'org.apache.hc', 'io.acryl.shaded.http'
-  relocate 'org.apache.commons.codec', 'datahub.spark2.shaded.o.a.c.codec'
-  relocate 'org.apache.commons.compress', 'datahub.spark2.shaded.o.a.c.compress'
-  relocate 'org.apache.commons.lang3', 'datahub.spark2.shaded.o.a.c.lang3'
+  relocate 'org.apache.commons.codec', 'io.acryl.shaded.org.apache.commons.codec'
```
Comment (david-leifker) [inline metadata-integration/java/acryl-spark-lineage/build.gradle:132]: This part doesn't make sense. The packages should continue to be separated into codec, compress, and lang3

File: metadata-integration/java/acryl-spark-lineage/build.gradle [hunk 2]
```diff
@@ -106,6 +108,8 @@ shadowJar {
     exclude(dependency {
       exclude_modules.contains(it.name)
     })
+    exclude(dependency("org.slf4j::"))
```
Comment (david-leifker) [inline metadata-integration/java/acryl-spark-lineage/build.gradle:111]: I don't think this is correct. I think its supposed to be `dependency("org.slf4j::*")`

**Review by david-leifker — APPROVED** [review]
Just a few minor tweaks to the gradle syntax, otherwise looks good.

File: metadata-integration/java/datahub-client/build.gradle [hunk 3]
```diff
@@ -78,6 +80,11 @@ shadowJar {
   // https://github.com/johnrengelman/shadow/issues/729
   exclude('module-info.class', 'META-INF/versions/**',
       '**/LICENSE', '**/LICENSE*.txt', '**/NOTICE', '**/NOTICE.txt', 'licenses/**', 'log4j2.*', 'log4j.*')
+  dependencies {
+    exclude(dependency("org.slf4j::"))
```
Comment (david-leifker) [inline metadata-integration/java/datahub-client/build.gradle:84]: I think these need to end in `*`

File: metadata-integration/java/acryl-spark-lineage/build.gradle (hunk 2 above)
Comment (treff7es) [reply metadata-integration/java/acryl-spark-lineage/build.gradle:111]: Actually, it works but I will change to yours as it is more expressive

Comment (treff7es) [reply metadata-integration/java/acryl-spark-lineage/build.gradle:111]: I tested and yours doesn't work but this one works.

**Pull request merged**